pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -324,7 +324,12 @@ func (c *Cluster) makeMonDaemonContainer(monConfig *monConfig) corev1.Container
 		WorkingDir:    config.VarLogCephDir,
 	}
 
-	if monConfig.Port != DefaultMsgr2Port {
+	bindaddr := controller.ContainerEnvVarReference(podIPEnvVar)
+	if monConfig.Port == DefaultMsgr2Port {
```

Comment:
I hope this is not meant for external cluster reconcile.

Reply:
Correct, this only applies to internal mode where mons are configured.